### PR TITLE
OOP school library: unit tests

### PR DESCRIPTION
**Pull request by HanningtoneM**
## objectives :green_circle: 
Write unit tests using Rspec.
Understand an existing codebase in a way that allows for adding tests.

## completed :heavy_check_mark: 

- Add unit tests to your classes:
- Book
- Person
- Student
- Teacher
- Classroom
- Rental
- all classes that you created while implementing the Decorator design pattern

### CODE REVIEW

File: spec/book_spec.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  end
+
+  it "should print the book\'s title" do
+    @book.title.should eql 'title'
+  end
+
+  it "should print the book\'s author" do
+    @book.author.should eql 'author'
+  end
+end
```
Comment (DammyShittu) [inline spec/book_spec.rb]: - [x] All public methods should be tested. Include test for the **add_rental** method in your book class.

File: spec/classroom_spec.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+describe Classroom do
+  before :each do
+    @classroom = Classroom.new('English')
+  end
+
+  it 'classroom label should be English' do
+    @classroom.label.should eql 'English'
+  end
+end
```
Comment (DammyShittu) [inline spec/classroom_spec.rb]: - [x] Include test for the **`add_student`** method in the classroom class

File: spec/person_spec.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  end
+
+  it "person\'s parent permission should be true" do
+    @person.parent_permission.should eq true
+  end
+
+  it 'can_use_services? should return true' do
+    @person.can_use_services?.should eq true
+  end
+end
```
Comment (DammyShittu) [inline spec/person_spec.rb]: - [x] Include test for the **`add_rental`** method in the person class

File: spec/student_spec.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  it "student\'s name should be Jack" do
+    @student.name.should eql 'Jack'
+  end
+  it "student\'s parent permission should be false" do
+    @student.parent_permission.should eq false
+  end
+  it 'can_use_services? should return true' do
+    @student.can_use_services?.should eq true
+  end
+end
```
Comment (DammyShittu) [inline spec/student_spec.rb]: - [x] Include test for the **`play_hooky`** and **`classroom`** methods.

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @HanningtoneM @Sahar-SE,

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/student_spec.rb (hunk 4 above)
Comment (HanningtoneM) [reply spec/student_spec.rb]: @DammyShittu thankyou for the review but  I have tested add rental in rental_spec and Student in student_spec I don't think your comments are clear on what I should please elaborate

**Review by danmainah — APPROVED** [review]
Hi @HanningtoneM and @Sahar-SE, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️ No linter errors
✔️ Github flow was correctly used

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**